### PR TITLE
IR-81: Create tasks for tickets which are not present in any TaskProv…

### PR DESCRIPTION
**Pull request by ishugupta92**
…ider

### CODE REVIEW

File: apps/retrospective/services/sprint.go [hunk 1]
```diff
@@ -308,19 +317,29 @@ func (service SprintService) SyncSprintMemberData(sprintMemberID string, indepen
 		ticketIDs = append(ticketIDs, timeLog.TaskID)
 	}
 
+	for _, ticketID := range ticketIDs {
```
Comment (anmolghosh) [inline apps/retrospective/services/sprint.go]: merge with above loop as discussed

File: apps/retrospective/services/sprint.go [hunk 2]
```diff
@@ -289,6 +296,7 @@ func (service SprintService) SyncSprintMemberData(sprintMemberID string, indepen
 		sprint.CurrentlySyncing = true
 		err = db.Save(&sprint).Error
 		if err != nil {
+			utils.LogToSentry(err)
 			return err
 		}
```
Comment (anmolghosh) [inline apps/retrospective/services/sprint.go:301]: fix `if sprint.StartDate == nil || sprint.EndDate == nil {` handling as discussed

**Review by anmolghosh — CHANGES_REQUESTED** [review]

File: apps/retrospective/services/sprint.go (hunk 2 above)
Comment (ishugupta92) [reply apps/retrospective/services/sprint.go:301]: Done

File: apps/retrospective/services/sprint.go (hunk 1 above)
Comment (ishugupta92) [reply apps/retrospective/services/sprint.go]: Done

**Review by anmolghosh — APPROVED** [review]

**Pull request merged**